src/structures/Message.js

```diff
@@ -172,6 +175,30 @@ class Message extends Base {
     } else if (data.member && this.guild && this.author) {
       this.guild.members.add(Object.assign(data.member, { user: this.author }));
     }
+
+    /**
+     * Flags that are applied to the message
+     * @type {Readonly<MessageFlags>}
+     */
+    this.flags = new MessageFlags(data.flags).freeze();
+
+    /**
+     * Reference data sent in a crossposted message.
+     * @typedef {Object} MessageReference
+     * @property {string} channelID ID of the channel the message was crossposted from
+     * @property {string} guildID ID of the guild the message was crossposted from
+     * @property {string} [messageID] ID of the message that was crossposted
```

Comment:
The property is guaranteed, it just can be null, so shouldn't it be that then?
```suggestion
     * @property {?string} messageID ID of the message that was crossposted
```

Reply:
![image](https://user-images.githubusercontent.com/17960496/65147598-27556b80-da27-11e9-9c5d-c136a3e5143e.png)

which per discord's docs:

![image](https://user-images.githubusercontent.com/17960496/65147628-39370e80-da27-11e9-8b68-76bec5e21a68.png)

So, no. The property isn't guaranteed. Well, it is for basically everything current but for the first crossposted messages, it ain't

Reply:
I just noticed guild_id is also... optional. Can you make your suggestion be for the guild one too, I'll commit it!